titan-compiler/coder.lua

```diff
@@ -447,18 +572,32 @@ generate_stat = function(stat)
     end
 end
 
--- @returns (statements, c_lvalue)
+typedecl.declare(coder, "coder", "Lvalue", {
```

Comment:
I'm afraid that this heavy modification in the ast inside the coder will get out of hand soon. The coder should be as dumb as possible.